core/lib/spree/name.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'active_support/core_ext/hash'
+
+module Spree
```

Comment:
A last-minute additional comment: what about naming this value object `Spree::Address::Name`? I think it less generic since `Name` could be a lot of things.

Reply:
Absolutely, it's safer with an additional namespace